docs/apache-airflow-providers-snowflake/connections/snowflake.rst

```diff
@@ -0,0 +1,76 @@
+.. Licensed to the Apache Software Foundation (ASF) under one
+    or more contributor license agreements.  See the NOTICE file
+    distributed with this work for additional information
+    regarding copyright ownership.  The ASF licenses this file
+    to you under the Apache License, Version 2.0 (the
+    "License"); you may not use this file except in compliance
+    with the License.  You may obtain a copy of the License at
+
+ ..   http://www.apache.org/licenses/LICENSE-2.0
+
+ .. Unless required by applicable law or agreed to in writing,
+    software distributed under the License is distributed on an
+    "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+    KIND, either express or implied.  See the License for the
+    specific language governing permissions and limitations
+    under the License.
+
+
+
+.. _howto/connection:snowflake:
+
+Snowflake Connection
+====================
+
+The Snowflake connection type enables integrations with Snowflake.
+
+Authenticating to Snowflake
+---------------------------
+
+Authenticate to Snowflake using the `Snowflake python connector default authentication
+<https://docs.snowflake.com/en/user-guide/python-connector-example.html#connecting-using-the-default-authenticator>`_.
+
+Default Connection IDs
+----------------------
+
+Hooks, operators, and sensors related to Snowflake use ``snowflake_default`` by default.
+
+Configuring the Connection
+--------------------------
+
+Login
+    Specify the snowflake username.
+
+Password
+    Specify the snowflake password.
+
+Host (optional)
+    Specify the snowflake hostname.
+
+Schema (optional)
+    Specify the snowflake schema to be used.
+
+Extra (optional)
+    Specify the extra parameters (as json dictionary) that can be used in the snowflake connection.
+    The following parameters are all optional:
+
+    * ``account``: Snowflake account name.
+    * ``database``: Snowflake database name.
+    * ``region``: Warehouse region.
+    * ``warehouse``: Snowflake warehouse name.
+    * ``role``: Snowflake role.
+    * ``authenticator``: To connect using OAuth set this parameter ``oath``
+    * ``private_key_file``: Specify the path to the private key file.
+    * ``session_parameters``: Specify `session level parameters
+      <https://docs.snowflake.com/en/user-guide/python-connector-example.html#setting-session-parameters>`_
```

Comment:
Is this section correct? See https://github.com/apache/airflow/pull/14724

Reply:
yes, I'm not sure how to handle this since users that are using this documentation to build a URI would want to know that all these fields need to be specified as extras. I am assuming users using the Airflow UI would see that some of these have their own field and understand what's going on. Maybe it would be best to add a note about this somewhere in the documentation?

Reply:
Agree with @sunkickr. This documentation is mostly for those who want to define the connnections manually rather than via UI. When you configure the connection via UI you have dedicated fields, but when you create it manually via cli or env variable, those need to be extras (the UI remaps values to extras). I think it's best to leave it as it is - users of Airlfow know how it works, and even if not, the UI fields for those "custom" connections have enough description/are clear enough. And in case you have those custom fields, extra is not even visible in the UI, so it is pretty clear what's going on I think.